src/Storages/StorageMergeTree.cpp

```diff
@@ -835,6 +845,8 @@ CancellationCode StorageMergeTree::killMutation(const String & mutation_id)
         }
     }
 
+    mutation_backoff_policy.removeFromFailedByVersion(mutation_version);
```

Comment:
1. Do we need the same for ReplicatedMergeTree?
2. Feels like it will not work because multiple mutations may be merged together and executed at once. For example, if there's a faulty mutation with version 42 and another mutation with version 137, then a part `all_1_2_3` can be mutated directly to `all_1_2_3_137` (and this will fail due to the mutation 42). So according to the `if (static_cast<UInt64>(result_part->part_info.mutation) == it->first)` condition, `addPartMutationFailure` will be called for the version 137, and killing the faulty mutation 42 will not reset the backoff. Please add the corresponding test. Consider removing all the logic around the mutation version for simplicity and resetting all backoffs when killing any mutation